net/ipv4/tcp_output.c

```diff
@@ -2342,8 +2348,9 @@ static bool tcp_can_coalesce_send_queue_head(struct sock *sk, int len)
 		if (unlikely(TCP_SKB_CB(skb)->eor) || tcp_has_tx_tstamp(skb))
 			return false;
 #ifdef CONFIG_SECURITY_TEMPESTA
-		/* Do not coalesce tempesta and not tempesta skbs. */
-		if (skb_tfw_tls_type(skb))
+		/* Do not coalesce tempesta skbs with tls type or set mark. */
+		if (skb_tfw_tls_type(skb)
+		    || (sock_flag(sk, SOCK_TEMPESTA) && skb->mark))
```

Comment:
Why can't we coalesce neighbor skbs, even if they have the same `mark`? It seems we should return `false` only if `skb->mark` is different from `mark` for previous skb..

Reply:
Yes fixed